src/cli/filter_arg.zig

```diff
@@ -0,0 +1,195 @@
+
+fn findWorkspaceMembers(allocator: std.mem.Allocator, log: *bun.logger.Log, workspace_map: *Package.WorkspaceMap, workdir_: []const u8) !void {
+    bun.JSAst.Expr.Data.Store.create(bun.default_allocator);
+    bun.JSAst.Stmt.Data.Store.create(bun.default_allocator);
+
+    defer {
+        bun.JSAst.Expr.Data.Store.reset();
+        bun.JSAst.Stmt.Data.Store.reset();
+    }
+
+    var workdir = workdir_;
+
+    while (true) : (workdir = std.fs.path.dirname(workdir) orelse break) {
+        const parent_trimmed = strings.withoutTrailingSlash(workdir);
+        var name_buf: bun.PathBuffer = undefined;
+        @memcpy(name_buf[0..parent_trimmed.len], parent_trimmed);
+        name_buf[parent_trimmed.len..name_buf.len][0.."/package.json".len].* = "/package.json".*;
+        name_buf[parent_trimmed.len + "/package.json".len] = 0;
+        const json_path = name_buf[0 .. parent_trimmed.len + "/package.json".len];
+
+        log.msgs.clearRetainingCapacity();
+        log.errors = 0;
+        log.warnings = 0;
+
+        const json_file = std.fs.cwd().openFileZ(
+            name_buf[0 .. parent_trimmed.len + "/package.json".len :0].ptr,
+            .{ .mode = .read_only },
+        ) catch continue;
+        defer json_file.close();
+
+        const json_stat_size = try json_file.getEndPos();
+        const json_buf = try allocator.alloc(u8, json_stat_size + 64);
+        defer allocator.free(json_buf);
+        const json_len = try json_file.preadAll(json_buf, 0);
+        const json_source = bun.logger.Source.initPathString(json_path, json_buf[0..json_len]);
+        const json = try json_parser.ParseJSONUTF8(&json_source, log, allocator);
+
+        const prop = json.asProperty("workspaces") orelse continue;
+
+        const json_array = switch (prop.expr.data) {
+            .e_array => |arr| arr,
+            .e_object => |obj| if (obj.get("packages")) |packages| switch (packages.data) {
+                .e_array => |arr| arr,
+                else => break,
+            } else break,
+            else => break,
+        };
+        const name_prop = json.asProperty("name") orelse continue;
+        const name = switch (name_prop.expr.data) {
+            .e_string => |n| n.data,
+            else => break,
+        };
+
+        _ = Package.processWorkspaceNamesArray(
+            workspace_map,
+            allocator,
+            log,
+            json_array,
+            &json_source,
+            prop.loc,
+            null,
+        ) catch |err| {
+            return err;
+        };
+
+        // add the root package to the workspace map, too
+        const entry = Package.WorkspaceMap.Entry{ .name = try allocator.dupe(u8, name), .version = null, .name_loc = bun.logger.Loc.Empty };
+        try workspace_map.insert(try allocator.dupe(u8, parent_trimmed), entry);
+
+        return;
+    }
+
+    // if we were not able to find a workspace root, try globbing for package.json files
+
+    var walker = Glob.BunGlobWalker{};
```

Comment:
this glob walker should ignore `node_modules` and other directories that pnpm ignores if they exist. I believe this is done through the `ignore_filter_fn`

Reply:
fixed in latest commit